osu.Game.Rulesets.Catch/UI/CatchHitExplosion.cs

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics.Containers;
+using osu.Game.Rulesets.Catch.Objects;
+using osu.Game.Rulesets.Judgements;
+using osuTK.Graphics;
+
+namespace osu.Game.Rulesets.Catch.UI
+{
+    /// <summary>
+    /// The base for hit explosion types used with <see cref="PoolableHitExplosion"/>.
+    /// </summary>
+    public abstract class CatchHitExplosion : CompositeDrawable, ICatchHitExplosion
+    {
+        public override bool RemoveWhenNotAlive => false;
+        public override bool RemoveCompletedTransforms => false;
+
+        public Color4 ObjectColour { get; set; }
+        public PalpableCatchHitObject HitObject { get; set; }
+        public JudgementResult JudgementResult { get; set; }
+        public float CatcherWidth { get; set; }
+        public float CatchPosition { get; set; }
+
+        public abstract void Animate();
+
+        public void RunAnimation()
+        {
+            var resultTime = JudgementResult.TimeAbsolute;
+
+            LifetimeStart = resultTime;
+
+            ApplyTransformsAt(double.MinValue, true);
+            ClearTransforms(true);
+
+            using (BeginAbsoluteSequence(resultTime))
+            {
+                Animate();
+            }
+
+            LifetimeEnd = LatestTransformEndTime;
+        }
+    }
+
+    /// <summary>
+    /// Provides properties hit explosions can use.
+    /// </summary>
+    public interface ICatchHitExplosion
```

Comment:
I'm not sure how to make sense of this interface. It's a bit weird and I don't think it really needs to exist, given the abstract class should suffice on its own.

Reply:
Yeah I was going to bring that up in the change request about adding { get; set; }, I didn't really think I needed the interface at all ever since I added the abstract class. Although the abstract class was added a while after I originally made the interface. I agree.